vllm/worker/neuron_model_runner.py

```diff
@@ -233,6 +247,39 @@ def prepare_model_input(
                                    sampling_metadata=sampling_metadata,
                                    multi_modal_kwargs=multi_modal_kwargs)
 
+    def _update_neuron_sampling_params(self, sampling_metadata):
+        # Update Neuron sampling parameters (GenerationConfig in Neuron)
+        current_sampling_params = self.model_config.sampling_params
+        assert current_sampling_params is not None, (
+            f"Failed to update sampling_params, "
+            f"current sampling params is {current_sampling_params}")
+
+        top_k = current_sampling_params.top_k.copy()
+        top_p = current_sampling_params.top_p.copy()
+        temperature = current_sampling_params.temperature.copy()
+        for index, sequence_group_to_sample in enumerate(
+                sampling_metadata.seq_groups):
+            top_k[index] = self._convert_to_neuron_top_k(
+                sequence_group_to_sample.sampling_params.top_k)
+            top_p[index] = sequence_group_to_sample.sampling_params.top_p
+            temperature[index] = \
+                sequence_group_to_sample.sampling_params.temperature
+
+        # Only update the sampling parameters if there is a change.
+        if (top_k != current_sampling_params.top_k
```

Comment:
wouldn't sampling params change only when a new sequence group enters a batch? it may be more efficient way of checking than going through all params one by one

Reply:
i like the idea of applying the change only when a new sequence group enters a batch.

Reply:
ack, now we will record the previous batch's request_ids and we only update the generation config if there is a change in the request_ids.